src/box/watcher.h

```diff
@@ -259,6 +259,25 @@ box_broadcast(const char *key, size_t key_len,
 void
 box_broadcast_fmt(const char *key, const char *format, ...);
 
+/**
+ * Returns the data attached to a notification key.
+ *
+ * @param key Name of the notification key
+ * @param key_len Length of the key name
+ * @param[out] end End of the key data
+ * @retval Key data
+ *
+ * The function never fails.
+ *
+ * If there's no data attached to the given notification key (box_broadcast()
+ * has never been called for this key), the function returns NULL.
+ *
+ * Note that the data returned by this function may be updated by a concurrent
+ * call to box_broadcast() so the caller must copy it if it intends to yield.
```

Comment:
JFYI: it makes the function not safe against Lua GC. While `luamp_decode()` is working in `lbox_watch_once()`, it might invoke `__gc` handlers in Lua which could do `box.broadcast()` for the same key. I am fine with keeping it as is though. Anyway our protection agains Lua GC is full of holes.

Reply:
You're right, I missed that. This issue should be pretty easy to fix though: we just need to copy the data to an area allocated with `malloc` before calling `luamp_decode`. However, I don't see a way to reproduce this issue i.e. the fix would be untested. So I prefer to leave it as is and fix it only if someone actually complains. Maybe, we should just disable Lua GC while in Lua C and call it only on return to Lua. This would fix most of these potential issues.

Reply:
The test would be simple - you just need to repeat what is done in `test/app-tap/gh-5632-6050-6259-gc-buf-reuse.test.lua`. I am fine with leaving it as is too.